frontend/src/app/forms/forms.page.ts

```diff
@@ -279,4 +319,63 @@ export class FormsPage implements OnInit, OnDestroy {
     }
     this.progressPercentage = validSteps / totalNumberOfRequiredSteps;
   }
+
+  /**
+   * Showing an alert message, when leaving a form page.
+   *
+   * returning a Promise<boolean>
+   * returning true, if user wants to leave the page.
+   * returning false, if user cancels leaving the page.
+   */
+  private async showClosingAlert(): Promise<boolean> {
+    const result = new Promise<boolean>(async (resolve) => {
+      const alert = await this.alertController.create({
+        header: 'Close',
+        message: 'Do you really want to cancel your job application?',
+        cssClass: 'custom-alert-button-colors',
+        buttons: [
+          {
+            text: 'No',
+            role: 'cancel',
+            handler: () => {
+              resolve(false);
+            },
+          },
+          {
+            text: 'Yes',
+            cssClass: 'color-secondary',
+            handler: () => {
+              resolve(true);
+            },
+          },
+        ],
+      });
+
+      alert.present();
+    });
+    return result;
+  }
+
+  /**
+   * Deciding Function to handle leaving forms page, needed for the Guard
+   *
+   * @param hasSubmittedForm Boolean to check, if the applicant wants to leave the forms page by submitting,
+   * if applicant wants to submit, mayLeaveView() return true, so Guard will allow leaving the page.
+   *
+   * returning true = Guard will allow leaving the page.
+   * returning false = Guard will stop leaving the page.
```

Comment:
This method does not receive a param. So please remove the `@param` lines.

And I would maybe replace the lines starting with "returning" with sth like:

```
@returns true (either directly or emitted in the Observable) if the view may be left. Otherwise, false is returned (or emitted).
```